doc/CASA.rst

```diff
@@ -0,0 +1,117 @@
+processing.
+
+CASA can provide most of the advantages of both those two options,
+while avoiding their drawbacks: it is a tradeoff between simplicity
+and agility.
+
+Specification
+=============
+
+Description
+-----------
+In CASA, every AS chooses exactly **one algorithm** per algorithm
+category. Those categories are globally defined and can either be
+general (PRF, MAC, ...) or protocol-specific (PRF-DRKEY,
+MAC-EPIC-HVF, ...).
+
+::
+
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | MAC:          | CMAC-AES    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | PRF:          | CBC-MAC-AES |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | MAC-EPIC-HVF: | Poly1305    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | PRF-DRKEY:    | CBC-MAC-AES |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | ...           | ...         |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+The idea behind the general categories is to cover a large set of
+protocols that only need algorithm agreement between different
+entities.
+On the other hand, the protocol-specific categories serve the needs
+of protocols that have additional requirements like high
+performance.
+In case an algorithm for a protocol-specific category is announced,
+it has priority over the one specified in the general category.
+
+Those algorithms are then announced in the beacons, so that this
+information is directly available and does not need to be fetched
+(from every on-path AS) by each protocol separately.
+
+Note the asymmetry in the CASA design:
+There is in general one side with higher and one with lower
+efficiency requirements. The side that requires higher efficiency
+chooses the algorithm to be used, as it is the one that is primarily
+affected by the algorithm's performance. This mirrors the design of
+DRKey, where one side has to fetch the key (slow), and the other side
+can recompute the key on the fly (fast).
+
+Beaconing Extensions
+--------------------
+For each category, the beacon needs to be extended by a field that
+indicates the concrete algorithm implementation that an AS supports.
+With a field size of one byte, there are 256 possible algorithms per
+category.
+
+For a scenario with ten different categories and a path consisting
+of eight ASes, the communication overhead introduced in the beacons
+due to CASA amounts to 80 bytes.
+
+Advantages
+----------
+With CASA, every AS is free to choose which cryptographic algorithms
+it wants to support, there is no global agreement necessary,
+except for the different categories and their options regarding the
+algorithms.
+Furthermore, CASA introduces only a small overhead in the beacons
+and does not affect the layout of the data plane packets in any way,
+meaning that no additional fields are necessary.
+The border routers of the ASes only need to implement the small
+number of algorithms promoted by their AS, which minimizes their
+overhead and prevents performance degradation.
+
+Algorithm Categories
```

Comment:
This should be extended and algorithms added. Also, explicitly add DRKey, EPIC, COLIBRI, SCMP.